android/src/main/java/com/applovin/reactnative/AppLovinMAXNativeAdView.java

```diff
@@ -108,19 +116,84 @@ public void setCustomData(@Nullable final String value)
         customData = value;
     }
 
-    public void setExtraParameters(@Nullable final ReadableMap readableMap)
+    public void setExtraParameters(@Nullable final ReadableArray readableArray)
     {
-        if ( readableMap != null )
+        if ( readableArray == null ) return;
+
+        Map<String, Object> extraParametersMap = new HashMap<>();
+
+        for ( int i = 0; i < readableArray.size(); i++ )
         {
-            extraParameters = readableMap.toHashMap();
+            ReadableMap map = readableArray.getMap( i );
+
+            String key = map.getString( "key" );
+
+            if ( !AppLovinSdkUtils.isValidString( key ) ) continue;
+
+            if ( map.hasKey( "value" ) )
+            {
+                Dynamic value = map.getDynamic( "value" );
+                ReadableType type = map.getType( "value" );
+
+                if ( type == ReadableType.String )
+                {
+                    extraParametersMap.put( key, value.asString() );
+                }
+                else if ( type == ReadableType.Null )
+                {
+                    extraParametersMap.put( key, null );
```

Comment:
If null - should remove? When is this going to be the case?

Reply:
Can null be used to reset the value of the key?